crypto/crypto-derive/src/lib.rs

```diff
@@ -368,6 +368,11 @@ pub fn hasher_dispatch(input: TokenStream) -> TokenStream {
         }
 
         impl libra_crypto::hash::CryptoHasher for #hasher_name {
+            fn salt() -> std::borrow::Cow<'static, str> {
+                std::borrow::Cow::from(libra_crypto::_serde_name::trace_name::<#type_name #param>()
```

Comment:
Where do we apply hashing in salts? This is required to ensure that no salt is the prefix of another salt.

Reply:
Yeah, we definitely need to enforce this in one way or another.

@matbd is this a guarantee we can enforce with tooling?

@sblackshear @davidiw How does the complexity increment from "pass these salt bytes and object bytes to EdDSA signing" to "pass these *hashed* bytes and object bytes to EdDSA signing" fare with our specification efforts?

Reply:
Currently, the *salt* is defined as `format!("{}::{}", LIBRA_HASH_PREFIX, serde_name)`.
We seed the hashing using `sha3_256(salt)` to avoid worrying about prefix-collisions and offer a simplified programmer experience. (*seed* = obligatory first bytes to hash before any LCS payload).

@huitseeker I suspect you want to expose the seed here. The return type is likely to be `&'static [u8]` as we probably want to memoize the seed.

Reply:
Hashing of salts is [there](https://github.com/libra/libra/pull/4073/files#diff-7d2098dbb70aee9e033b266567340498R382-R392) and [there](https://github.com/libra/libra/pull/4073/files#diff-040d5f23675cd0d3adaa822575b670edR523-R529).